Xamarin.Essentials/Preferences/Preferences.win32.cs

```diff
@@ -0,0 +1,133 @@
+                {
+                    try
+                    {
+                        var formatter = new BinaryFormatter();
+                        var readPreferences = (Dictionary<string, Dictionary<string, object>>)formatter.Deserialize(stream);
+                        stream.Close();
+
+                        if (readPreferences is object)
+                        {
+                            preferences = readPreferences;
+                        }
+                    }
+                    catch (SerializationException)
+                    {
+                        // if deserialization fails proceed with empty settings
+                    }
+                }
+            }
+            else
+            {
+                if (!Directory.Exists(Path.GetDirectoryName(settingsPath)))
+                {
+                    // create folder for app settings
+                    Directory.CreateDirectory(Path.GetDirectoryName(settingsPath));
+                }
+            }
+
+            if (!preferences.ContainsKey(string.Empty))
+            {
+                preferences.Add(string.Empty, new Dictionary<string, object>());
+            }
+        }
+
+        static void Save()
+        {
+            using (var stream = File.OpenWrite(settingsPath))
+            {
+                var formatter = new BinaryFormatter();
+                formatter.Serialize(stream, preferences);
+                stream.Close();
+            }
+        }
+
+        static string CleanSharedName(string sharedName)
+        {
+            if (string.IsNullOrEmpty(sharedName))
+            {
+                return string.Empty;
+            }
+
+            return sharedName;
+        }
+
+        static bool PlatformContainsKey(string key, string sharedName)
+        {
+            if (preferences.ContainsKey(CleanSharedName(sharedName)))
+            {
+                return preferences[sharedName].ContainsKey(key);
+            }
+
+            return false;
+        }
+
+        static void PlatformRemove(string key, string sharedName)
+        {
+            if (preferences.ContainsKey(CleanSharedName(sharedName)))
+            {
+                preferences[CleanSharedName(sharedName)].Remove(key);
+                Save();
+            }
+        }
+
+        static void PlatformClear(string sharedName)
+        {
+            if (preferences.ContainsKey(CleanSharedName(sharedName)))
+            {
+                preferences[CleanSharedName(sharedName)].Clear();
+                Save();
+            }
+        }
+
+        static void PlatformSet<T>(string key, T value, string sharedName)
+        {
+            if (!preferences.ContainsKey(CleanSharedName(sharedName)))
+            {
+                preferences.Add(CleanSharedName(sharedName), new Dictionary<string, object>());
+            }
+
+            if (preferences[CleanSharedName(sharedName)].ContainsKey(key))
```

Comment:
Maybe use a TryGetValue for these.

In fact, just use the `[key] = value` here as it will add any missing keys. No need to check and then branch code.